### PR TITLE
Add capability to enable MRC histogram generation on a pool

### PR DESCRIPTION
**Pull request by lpl6338236**
Add librados APIs and rados command line tool to get MRCs.
Add "ceph osd pool set/get“ command line tool to manipulate mrc parameters.

### CODE REVIEW

**Comment by tchaikov** [comment]
@lpl6338236 iiuc, your change offers us new metrics, MRC, to analyse the cache performance of a pool. so to offer the broader reviewers more context, i am updating your pull request's title accordingly.

File: src/osd/ReuseDist.h [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+  public:
+    node(const T &k,COLOR c = red)
+      : key(k), color(c), parent(NULL),
+      left(NULL), right(NULL), sum(0) {}
+
+};
+
+
+template <typename T>
+class RBTree
```
Comment (yuyuyu101) [inline src/osd/ReuseDist.h]: It's cool to implment RBTree here, I'd like to move these to a new file named RBTree.h

Comment (yuyuyu101) [inline src/osd/ReuseDist.h]: I also implement a RBTree at https://github.com/yuyuyu101/ceph/blob/librbd-cache/src/common/RBTree.h .
maybe you can employ some tests from here https://github.com/yuyuyu101/ceph/blob/librbd-cache/src/test//common/test_rbtree.cc

Comment (liewegas) [inline src/osd/ReuseDist.h]: > It's cool to implment RBTree here, I'd like to move these to a new file named RBTree.h

Can we use boost::intrusive::set<> instead?

**Comment by liewegas** [comment]
Closing this pull request since it appears to be abandoned.  If you it is not abandoned, and you are still interested in championing this change, please address any review comments so far and reopen the pull request.

**Pull request closed without merging**